File: src/main/java/no/nav/bidrag/beregn/barnebidrag/BarnebidragCore.java
package no.nav.bidrag.beregn.barnebidrag;

import no.nav.bidrag.beregn.barnebidrag.dto.BeregnBarnebidragGrunnlagCore;
import no.nav.bidrag.beregn.barnebidrag.dto.BeregnetBarnebidragResultatCore;
import no.nav.bidrag.beregn.barnebidrag.periode.BarnebidragPeriode;


public interface BarnebidragCore {

  BeregnetBarnebidragResultatCore beregnBarnebidrag(BeregnBarnebidragGrunnlagCore beregnBarnebidragGrunnlagCore);

  static BarnebidragCore getInstance() {
    return new BarnebidragCoreImpl(BarnebidragPeriode.Companion.getInstance());
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/barnebidrag/BarnebidragCore.kt
package no.nav.bidrag.beregn.barnebidrag

import no.nav.bidrag.beregn.barnebidrag.dto.BeregnBarnebidragGrunnlagCore
import no.nav.bidrag.beregn.barnebidrag.dto.BeregnetBarnebidragResultatCore
import no.nav.bidrag.beregn.barnebidrag.periode.BarnebidragPeriode

fun interface BarnebidragCore {
    fun beregnBarnebidrag(grunnlag: BeregnBarnebidragGrunnlagCore): BeregnetBarnebidragResultatCore

    companion object {
        fun getInstance(): BarnebidragCore = BarnebidragCoreImpl(BarnebidragPeriode.getInstance())
    }
}
